### PR TITLE
Feature/update kwt

### PR DESCRIPTION
**Pull request by martynpclark**
Updating code to include kinematic wave routing

### CODE REVIEW

File: route/build/src/network_topo.f90 [hunk 1]
```diff
@@ -655,6 +660,14 @@ SUBROUTINE REACH_LIST(&
   structSEG(iRch)%var(ixSEG%totalArea)%dat(1) = structSEG(iRch)%var(ixSEG%basArea)%dat(1) + &
                                                 structSEG(iRch)%var(ixSEG%upsArea)%dat(1)
 
+  ! ---------- define good basins ---------------------------------------------------------
+
```
Comment (nmizukami) [inline route/build/src/network_topo.f90]: i think need to define what the good basin is here.   
"goodBasin"  == Drainage area at the bottom of the reach is greater than threshold, "verySmall" value.

File: route/build/src/network_topo.f90 [hunk 2]
```diff
@@ -548,6 +551,7 @@ SUBROUTINE REACH_LIST(&
  integer(i4b)      , intent(out)                :: ierr            ! error code
  character(*)      , intent(out)                :: message         ! error message
  ! local variables
+ real(dp), parameter                            :: verySmall=tiny(1.0_dp) ! a very small number
```
Comment (nmizukami) [inline route/build/src/network_topo.f90]: verySmall is defined in public.f90.   

Now, not clear what to be stored in public.f90, globalData.f90.    maybe have to have some rule.

File: route/build/src/process_ntopo.f90 [hunk 3]
```diff
@@ -65,6 +68,7 @@ subroutine ntopo(&
  ! --------------------------------------------------------------------------------------------------------------
  ! local variables
  character(len=strLen)           :: cmessage           ! error message of downwind routine
+ real(dp), parameter             :: min_slope=1.e-6_dp ! minimum slope
```
Comment (nmizukami) [inline route/build/src/process_ntopo.f90]: Should this be in public.f90?

File: route/build/src/kwt_route.f90 [hunk 4]
```diff
@@ -25,7 +25,7 @@ module kwt_route
 public::qroute_rch
 
 ! common variables
-integer(i4b),parameter  :: ixPrint = -9999  ! the desired reach (set to negative to avoid any printing)
+integer(i4b),save       :: ixPrint = -9999  ! the desired reach (set to negative to avoid any printing)
 real(dp),parameter      :: verySmall=tiny(1.0_dp)  ! a very small number
```
Comment (nmizukami) [inline route/build/src/kwt_route.f90]: defined in public.f90?

File: route/build/src/kwt_route.f90 [hunk 5]
```diff
@@ -112,8 +98,9 @@ subroutine QROUTE_RCH(IENS,JRCH,    & ! input: array indices
    implicit none
    ! Input
    INTEGER(I4B), INTENT(IN)                    :: IENS          ! ensemble member
-   integer(i4b), intent(in)                    :: ixOutlet      ! index of the outlet reach
    INTEGER(I4B), INTENT(IN)                    :: JRCH          ! reach to process
+   integer(i4b), intent(in)                    :: ixDesire      ! index of the desired reach
```
[the remark this answers is not in the document]
Comment (martynpclark) [reply route/build/src/kwt_route.f90]: Fixed

File: route/build/src/network_topo.f90 (hunk 1 above)
Comment (martynpclark) [reply route/build/src/network_topo.f90]: Fixed

File: route/build/src/network_topo.f90 (hunk 2 above)
Comment (martynpclark) [reply route/build/src/network_topo.f90]: Fixed

File: route/build/src/process_ntopo.f90 (hunk 3 above)
Comment (martynpclark) [reply route/build/src/process_ntopo.f90]: Fixed

File: route/build/src/kwt_route.f90 (hunk 4 above)
Comment (martynpclark) [reply route/build/src/kwt_route.f90]: Fixed

**Comment by martynpclark** [comment]
Addressed all comments raised during code review.

**Pull request merged**